docs/DeveloperGuide.md

```diff
@@ -1,37 +1,209 @@
+
+There are three possible types of tasks - Todo, Deadline and Repeat.
+
+The sequence to add tasks involves:
+1. **User Input**:
+* **Todo**: The user enters the `todo` command followed by tag details (e.g., /tag).
+* **Deadline**: The user enters the `deadline` command followed by the deadline (e.g., /by) and tag details (e.g., /tag).
+* * **Repeat**: The user enters the `repeat` command followed by the recurring basis (e.g., /every) and tag details (e.g., /tag).
+2. **Command Parsing**:
+* **Todo Task**: The `Parser` parses the input and creates an `AddTodoParser` object.
+* **Deadline Task**: The `Parser` parses the input and creates an `AddDeadlineParser` object.
+* **Repeat Task**: The `Parser` parses the input and creates an `AddRepeatParser` object.
+3. **Execution**: The `AddTaskCommand` adds the task.
+4. **Storage Update**: The updated patient's data, now containing the new task, is saved to storage.
+
+
+
+#### Sequence Diagram
+
+The following sequence diagram illustrates how `Parser` parse command for `AddTaskCommand`:
+
+![Parse_Add_Task_Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/ParseAddTaskClassDiagram.png)
+
+A closer look on how `AddTaskCommand` is executed is shown below:
+
+![Add_Task_Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/AddTaskClassDiagram.png)
+
+#### Implementation considerations:
+The `AddTaskCommand` has been implemented with a design that prioritizes scalability and maintainability. This allows the system to easily support additional task types in the future without requiring major code changes. For example, if new task types are needed in the future (e.g., `event`, `appointment`), they can be added by simply extending the Task class hierarchy and updating the createTask() factory method.
+### **FindCommand**
+
+The find feature allows users to find the name of a patient, or the name of a task. This is facilitated by the `FindPatientCommand` and `FindTaskCommand`.
+
+1. **User Input**: The user enters the `find` command followed by keywords
+2. **Command Parsing**: The `Parser` parses the input and creates a `FindParser` object.
+3. **Execution**: The `FindPatientCommand` or `FindTaskCommand` searches for the task.
+
+#### Sequence Diagram
+
+The following sequence diagram illustrates how the `FindPatientCommand` is executed:
+
+![Find_Sequence_Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/FindSequenceDiagram.png)
+
+### **State Switching Feature**
+
+The state switching feature allows the system to change its mode of operation between `MAIN_STATE` (for patient management) and `TASK_STATE` (for task management). This allows the software to handle commands differently based on the current state.
```

Comment:
Usage of state management is explained very clearly, making it easy to understand!